internal/exec/stages/files/units.go

```diff
@@ -17,57 +17,145 @@ package files
 import (
 	"fmt"
 	"path/filepath"
+	"strconv"
 	"strings"
 
+	"github.com/coreos/go-systemd/dbus"
 	"github.com/coreos/ignition/v2/config/v3_1_experimental/types"
 	"github.com/coreos/ignition/v2/internal/distro"
 	"github.com/coreos/ignition/v2/internal/exec/util"
 )
 
+type Preset struct {
+	unit            string
+	enabled         bool
+	instantiateable bool
+	instances       []string
+}
+
+// isRequiredSystemdVersionExists checks the version of Systemd
+// in a given FCOS/RHCOS system.
+func (s *stage) isRequiredSystemdVersionExists() error {
+	var systemdVersion string
+	conn, err := dbus.New()
+	if err != nil {
+		return err
+	}
+	version, err := conn.GetManagerProperty("Version")
+	if err != nil {
+		return err
+	}
+	//for FCOS, the version of systemd is printed like 'v243.5-1.fc31'
```

Comment:
let's make this slightly more generic and not reference FCOS/RHCOS here but probably more likely Fedora and RHEL8.

So we could have a higher level comment here like:

```
// Handle different systemd versioning schemes that are being returned.
// For example:
//    - Fedora 31: `v243.5-1.fc31`
//    -    RHEL 8: `239`
```

So we need to handle the `v` on the front, but also the `.`, but let's try to do those independently because Fedora and RHEL aren't the only two cases we have to ever consider. Something like:

```
package main
import "fmt"
import "strings"

func main() {
    var version string
    var systemdVersion string
    var begin int
    var end int
    version = "v243.5-1.fc31"
    version = "239"
    version = "42.21"

    // Handle different systemd versioning schemes that are being returned.
    // For example:
    //    - Fedora 31: `v243.5-1.fc31`
    //    -    RHEL 8: `239`
    //
    // If we encounter a prefixed `v` trim it off
    // If we encounter a `.` trim everything after it
    if strings.HasPrefix(version, "v") {
        begin = 1
    } else {
        begin = 0
    }
    end = strings.Index(version, ".")
    if end == -1 {
        end = len(version)
    }
    systemdVersion = version[begin:end]
    fmt.Println(systemdVersion)
}
```

Reply:
Ack